docs/source/squirrel/cpp_api/sq_functions.rst

```diff
@@ -0,0 +1,211 @@
+    }
+
+Return a complex ``ornull`` type:
+
+.. code-block:: cpp
+
+    ADD_SQFUNC("int ornull", CPlugComplex, "int n", "returns null", ScriptContext::CLIENT | ScriptContext::SERVER | ScriptContext::UI)
+    {
+        SQInteger n = g_pSquirrel<context>->getinteger(sqvm, 1);
+        
+        if (n == 0)
+            return SQRESULT_NULL;
+
+        g_pSquirrel<context>->newarray(sqvm);
+        g_pSquirrel<context>->pushinteger(sqvm, n);
+        g_pSquirrel<context>->arrayappend(sqvm, 2);
+        g_pSquirrel<context>->pushinteger(sqvm, n * 2);
+        g_pSquirrel<context>->arrayappend(sqvm, 2);
+
+        return SQRESULT_NOTNULL; // return the array [ n, n * 2 ]
+    }
+
+Replacing Squirrel Functions
+----------------------------
+
+You can use the ``REPLACE_SQFUNC`` macro to replace an existing sq function.
+
+.. cpp:function:: macro REPLACE_SQFUNC(funcName, runOnContext)
+
+    :param funcName: The name of the function to replace
+    :param runOnContext: The contexts that have access to this function
+
+It's also possible to add an override directly with the ``AddFuncOverride`` function of the ``SquirrelManager`` class.
+
+.. cpp_function:: void AddFuncOverride(std::string name, SQFunction func)
+
+    :param std::string name: The name of the function to override
+    :param SQFunc func: A function object that replaces the logic
+
+.. code-block:: cpp
+
+    // Replaces dangerous vanilla functions to only log their call with no further logic.
+    g_pSquirrel<context>->AddFuncOverride("DevTextBufferWrite", SQ_StubbedFunc<context, "DevTextBufferWrite">);
+    g_pSquirrel<context>->AddFuncOverride("DevTextBufferClear", SQ_StubbedFunc<context, "DevTextBufferClear">);
+    g_pSquirrel<context>->AddFuncOverride("DevTextBufferDumpToFile", SQ_StubbedFunc<context, "DevTextBufferDumpToFile">);
+
+Script Contexts
+---------------
+
+Scriptcontexts are used to define the VMs that have access to a native function. Available Contexts are
+
+- ``ScriptContext::SERVER`` - The SERVER sqvm
+- ``ScriptContext::CLIENT`` - The CLIENT sqvm
+- ``ScriptContext::UI`` - The UI vm
+
+Script Returns
+--------------
+
+Squirrel functions need to return a ``SQRESULT``. Valid results are
+
+- ``SQRESULT_NULL`` - This function returns ``null``. Nothing is left over on the stack.
+- ``SQRESULT_NOTNULL`` - This functions returns the last item on the stack.
+- ``SQRESULT_ERROR`` - This function has thrown an error.
+
+Calling
+-------
+
+.. _Call:
+
+.. cpp:function:: SQRESULT Call(const char* funcname)
+
+    :param char* funcname: Name of the function to call
+    
+    This function assumes the squirrel VM is stopped/blocked at the moment of call
+
+    Calling this function while the VM is running is likely to result in a crash due to stack destruction
+
+    If you want to call into squirrel asynchronously, use `AsyncCall`_ instead.
+
+    .. code-block:: cpp
+
+        Call("PluginCallbackTest"); // PluginCallbackTest()
+
+.. _Call_args:
```

Comment:
why `_Call_args`?

Reply:
Call has an override that pushes arguments to the stack for you.

Labels can't be duplicates and are only used for linking internally, so the user won't see them.